plugin/storage/factory.go

```diff
@@ -134,23 +140,49 @@ func (*Factory) getFactoryOfType(factoryType string) (storage.Factory, error) {
 	}
 }
 
+func (*Factory) getArchiveFactoryOfType(factoryType string) (storage.Factory, error) {
+	switch factoryType {
+	case cassandraStorageType:
+		return cassandra.NewArchiveFactory(), nil
+	case elasticsearchStorageType, opensearchStorageType:
+		return es.NewArchiveFactory(), nil
+	default:
+		return nil, fmt.Errorf("archive storage factory not available for %s", factoryType)
+	}
+}
+
 // Initialize implements storage.Factory.
 func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger) error {
 	f.metricsFactory = metricsFactory
-	for kind, factory := range f.factories {
-		mf := metricsFactory.Namespace(metrics.NSOptions{
-			Name: "storage",
-			Tags: map[string]string{
-				"kind": kind,
-				"role": "primary", // can be overiden in the storage factory for archive/sampling stores
-			},
-		})
-		if err := factory.Initialize(mf, logger); err != nil {
-			return err
+	initializeFactories := func(factories map[string]storage.Factory, role string) error {
+		for kind, factory := range factories {
+			mf := metricsFactory.Namespace(metrics.NSOptions{
+				Name: "storage",
+				Tags: map[string]string{
+					"kind": kind,
+					"role": role,
+				},
+			})
+			if err := factory.Initialize(mf, logger); err != nil {
+				return err
+			}
+			if role == "archive" {
+				if primaryFactory, ok := f.factories[kind]; ok {
+					if dc, ok := factory.(plugin.DefaultConfigurable); ok {
+						dc.InheritSettingsFrom(primaryFactory)
+					}
+				}
```

Comment:
is the coverage gap here real or stale?
<img width="547" alt="image" src="https://github.com/user-attachments/assets/d7e1c58e-2108-4ca9-b3e9-2505a70fa47d" />